api/src/server.ts

```diff
@@ -52,6 +59,21 @@ const start = async () => {
       credentials: true,
     },
   });
+
+  io.on('connection', async socket => {
+    socket.on('meeting:join', meetingId => {
+      const { principalId } = socket.request.session;
+      if (participantExists(meetingId, principalId)) {
+        socket.join(`meeting:${meetingId}`);
+        socket.emit('meeting:joined', meetingId);
+      }
+    });
+    socket.on('meeting:leave', meetingId => {
+      socket.leave(`meeting:${meetingId}`);
```

Comment:
Can you help me understand why leaving the room doesn't require checking who the participant is?

Reply:
I personally do not think we need to check the participant 2nd time upon room leaving. The major point is to check for security purposes participant identity and relevance to the meeting upon joining the room.

Reply:
In the event that someone is removed as a participant from a meeting while they are connected, if it checked whether they were still a participant, then they would be forever stuck in a room for a meeting they aren't associated to.

Anyway, even if that case never happens, there's no value to doing the extra work of checking. If a client wants to leave a room, why not just let them?

Reply:
Okay I see, thanks for the clarifications!